core/include/scipp/core/element/comparison.h

```diff
@@ -15,19 +15,6 @@
 /// operations for Variable.
 namespace scipp::core::element {
 
-constexpr auto comparison = overloaded{
-    transform_flags::no_out_variance,
-    arg_list<double, float, int64_t, int32_t, bool,
-             std::tuple<int64_t, int32_t>, std::tuple<int32_t, int64_t>>,
-    [](const units::Unit &x, const units::Unit &y) {
-      expect::equals(x, y);
-      return units::dimensionless;
-    }};
-
-constexpr auto less = overloaded{
-    comparison,
-    [](const auto &x, const auto &y) { return x < y; },
-};
 constexpr auto is_approx = overloaded{
     transform_flags::no_out_variance,
     arg_list<double, float, int64_t, int32_t, std::tuple<double, double, float>,
```

Comment:
is_approx, should probably be updated along the same lines, this could be done as a follow up.